packages/service/src/ocr/commands/ocrSetValue.ts

```diff
@@ -0,0 +1,62 @@
+import type { OcrSetValueOptions } from '../types.js'
+import ocrKeys from '../utils/ocrKeys.js'
+import ocrClickOnText from './ocrClickOnText.js'
+
+export default async function ocrSetValue(options: OcrSetValueOptions): Promise<void> {
+    const {
+        contrast,
+        clickDuration,
+        haystack,
+        fuzzyFindOptions,
+        isTesseractAvailable,
+        language,
+        ocrImagesPath,
+        submitValue = false,
+        text,
+        value,
+    } = options
+
+    // 1. First click on the position of the text to make sure it is intractable
+    await ocrClickOnText({
+        contrast,
+        clickDuration,
+        haystack,
+        fuzzyFindOptions,
+        isTesseractAvailable,
+        language,
+        ocrImagesPath,
+        text,
+    })
+
+    // 2. If Mobile then a keyboard might be shown
+    if (driver.isMobile) {
+        try {
+            // Wait for 3 seconds for the keyboard to be shown
+            await driver.waitUntil(
+                async () => driver.isKeyboardShown(),
+                {
+                    timeout: 3 * 1000,
+                    timeoutMsg: 'Keyboard was not hidden',
```

Comment:
I am not sure if the custom time out message makes sense if we ignore the catch block

Reply:
You're right